utils/geo.py

```
@@ -203,5 +202,21 @@ def get_best_sections_full_coverage(
     full_coverage = full_coverage.to_crs(epsg=4326)
 
     full_coverage = explode_mp(full_coverage)
+    full_coverage.geometry = full_coverage.geometry.buffer(0)
 
     return full_coverage
+
+
+def coverage_percentage(aoi: GDF, full_coverage: GDF) -> float:
+    """
+    Calculates percentage of coverage of AOI from full_coverage.
+    """
+
+    centroid = box(*full_coverage.total_bounds).centroid
+    epsg = get_utm_zone_epsg(lon=centroid.x, lat=centroid.y)
+    full_coverage = full_coverage.to_crs(epsg=epsg)
+    aoi = aoi.to_crs(epsg=epsg)
+
+    unary_union = gpd.GeoSeries(cascaded_union(full_coverage.geometry))
+    cov = (unary_union.area / aoi.area) * 100
```

Not likely case but maybe throw an error etc. if aoi df has more than one element, otherwise cov calculation would crash or return a series I guess. Or do aoi.iloc[0].area   etc.

---

If the AOI has several parts wouldn't it fail before anyway? Or you handled this cases in other functions?

---

Hm yeah you are right, this is pretty deep down. I don't think we are handling that case.

---

Leaving it as is, will write a tech debt ticket for this.

---

https://up42.atlassian.net/browse/UP-4762